test/tracing/test_set_direction.py

```diff
@@ -0,0 +1,76 @@
+#  Copyright 2021 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from zepben.evolve import TestNetworkBuilder, Feeder, ConductingEquipment, set_direction, FeederDirection, NetworkService
+
+
+@pytest.mark.asyncio
+async def test_set_direction():
+    network = _get_single_feeder_network()
+    await set_direction().run(network)
+
+    assert network.get("b0-t1").normal_feeder_direction == FeederDirection.NONE
+    assert network.get("b0-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+    assert network.get("c1-t1").normal_feeder_direction == FeederDirection.UPSTREAM
+    assert network.get("c1-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+    assert network.get("j2-t1").normal_feeder_direction == FeederDirection.UPSTREAM
+    assert network.get("j2-t2").normal_feeder_direction == FeederDirection.DOWNSTREAM
+
+
+@pytest.mark.asyncio
+async def test_set_direction_doesnt_flow_through_feeder_heads():
+    network = _get_dual_parallel_feeders_network()
+    await set_direction().run(network)
+
+    assert network.get("b0-t1").normal_feeder_direction == FeederDirection.NONE
+    assert network.get("b0-t2").normal_feeder_direction == FeederDirection.BOTH
+    assert network.get("c1-t1").normal_feeder_direction == FeederDirection.BOTH
+    assert network.get("c1-t2").normal_feeder_direction == FeederDirection.BOTH
+    assert network.get("b2-t1").normal_feeder_direction == FeederDirection.BOTH
+    assert network.get("b2-t2").normal_feeder_direction == FeederDirection.NONE
+
+
+# feeder_heads:     feeder
+#                   v
+# network:          b0 -- c1 -- j2
+def _get_single_feeder_network() -> NetworkService:
+    network = TestNetworkBuilder().from_breaker().to_acls().to_junction().network
+    feeder = Feeder(mrid="f")
```

Comment:
Cheers, @charlta , I updated the code to use 'add_feeder' instead of doing that work manually.
I updated the 'build' method to assign equipment to feeders (if there are any feeders present) because it did not previously, my guess is this was just missed given your comment below saying that it should set feeders.